profiler/profiler.go

```diff
@@ -164,33 +164,37 @@ func (s *Server) EnableProfiler(portNumber int32) (string, error) {
 	}
 
 	profilerAddr := fmt.Sprintf(":%d", profilerPort)
-	s.server = &http.Server{
+	newServer := &http.Server{
 		Addr:              profilerAddr,
 		ReadHeaderTimeout: 10 * time.Second,
 	}
 	go func() {
-		if err := s.server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
-			logrus.WithError(err).Warnf("Get error when start profiler server %v", s.server.Addr)
-			s.server = nil
+		if err := newServer.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+			logrus.WithError(err).Warnf("Get error when start profiler server %v", newServer.Addr)
 			s.errMsg = err.Error()
 			return
 		}
-		logrus.Infof("Profiler server (%v) is closed", s.server.Addr)
+		logrus.Infof("Profiler server (%v) is closed", newServer.Addr)
 	}()
 
-	logrus.Infof("Waiting the profiler server(%v) to start", s.server.Addr)
+	logrus.Infof("Waiting the profiler server(%v) to start", newServer.Addr)
 	// Wait for the profiler server to start, and check the profiler server.
 	retryCount := 3
 	for i := 0; i < retryCount; i++ {
-		conn, err := net.DialTimeout("tcp", s.server.Addr, 1*time.Second)
+		conn, err := net.DialTimeout("tcp", newServer.Addr, 1*time.Second)
 		if err == nil {
 			_ = conn.Close()
 			break
 		}
```

Comment:
What if there is a timeout? This function does not check or handle `err` after this retry loop.

Reply:
Hi @shuo-wu,
Here is the checking for the profiler server.
And if the timeout, the conn is not alive, and we will retry another time (total three times).
If that all fails, we assume the profiler server is not enabled.

Then, we will close the profiler server and return the error.

Reply:
Addressed. @Vicente-Cheng PTAL.